src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -887,6 +887,13 @@ private static bool AddFlags(TypeSymbol type, ArrayBuilder<bool> transformFlagsB
                         transformFlagsBuilder.Add(false);
                         break;
 
+                    case TypeKind.FunctionPointer:
+                        handleFunctionPointerType((FunctionPointerTypeSymbol)type, transformFlagsBuilder, isNestedNamedType, addCustomModifierFlags);
+
+                        // Function pointer types have nested custom modifiers and refkinds in line with types, and visit all their nested types
+                        // as part of this call.
+                        return true;
```

Comment:
>return true; [](start = 24, length = 12)

Could returning true cut the visit process short and leave some constituent parts unvisited?

Reply:
Yeah, this is busted for things like `D<delegate*<dynamic>[], dynamic>`. I'll need to think on this some more. Fundamentally we need to visit more than `VisitTypes` normally will visit here, and in a different order, but we also want to keep going after this type is passed.

---
In reply to: [422527325](https://github.com/dotnet/roslyn/pull/44078#discussion_r422527325) [](ancestors = 422527325)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/44160](https://github.com/dotnet/roslyn/issues/44160) to track. This is going to be too big a change to handle in this PR, and I think this is a niche enough case (and only results in dynamic not being treated as dynamic) that I'd prefer to address later.

---
In reply to: [423335853](https://github.com/dotnet/roslyn/pull/44078#discussion_r423335853) [](ancestors = 423335853,422527325)